src/panels/lovelace/cards/hui-map-card.js

```diff
@@ -0,0 +1,258 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+import Leaflet from 'leaflet';
+
+import '../../map/ha-entity-marker.js';
+
+import setupLeafletMap from '../../../common/dom/setup-leaflet-map.js';
+import processConfigEntities from '../common/process-config-entities.js';
+import computeStateDomain from '../../../common/entity/compute_state_domain.js';
+import computeStateName from '../../../common/entity/compute_state_name.js';
+import debounce from '../../../common/util/debounce.js';
+
+Leaflet.Icon.Default.imagePath = '/static/images/leaflet';
+
+class HuiMapCard extends PolymerElement {
+  static get template() {
+    return html`
+      <style>
+        :host([is-panel]) ha-card {
+          left: 0;
+          top: 0;
+          width: 100%;
+          /**
+           * In panel mode we want a full height map. Since parent #view 
+           * only sets min-height, we need absolute positioning here
+           */
+          height: 100%;
+          position: absolute;
+        }
+        
+        #map {
+          z-index: 0;
+          border: none;
+          position: absolute;
+          top: 0;
+          left: 0;
+          width: 100%;
+          height: 100%;
+        }
+        
+        #root {
+          width: 100%;
+          position: relative;
+        }
+        
+        :host([is-panel]) #root {
+          height: 100%;
+        }
+      </style>
+      
+      <ha-card id="card" header="[[_config.title]]">
+        <div id="root">
+          <div id="map"></div>
+        </div>
+      </ha-card>
+
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+        observer: '_drawEntities'
+      },
+      _config: Object,
+      isPanel: {
+        type: Boolean,
+        reflectToAttribute: true
+      }
+    };
+  }
+
+  constructor() {
+    super();
+    this._debouncedResizeListener = debounce(this._resetMap.bind(this), 100);
+  }
+
+  ready() {
+    super.ready();
+
+    if (!this._config || this.isPanel) {
+      return;
+    }
+
+    this.$.root.style.paddingTop = this._config.aspect_ratio || '100%';
+  }
+
+  setConfig(config) {
+    if (!config) {
+      throw new Error('Error in card configuration.');
+    }
+
+    this._configEntities = processConfigEntities(config.entities);
+    this._config = config;
+  }
+
+  getCardSize() {
+    let ar = this._config.aspect_ratio || '100%';
+    ar = ar.substr(0, ar.length - 1);
+    return 1 + Math.floor(ar / 25) || 3;
+  }
+
+  connectedCallback() {
```

Comment:
Please add a `disconnectedCallback` that removes the map: [docs](https://leafletjs.com/reference-1.3.0.html#map-remove)

Reply:
Do you want me to add this to the old map panel as well, or does it not matter there?

Reply:
Yes please do.